### PR TITLE
refactor(ts): enable noImplicitAny on chunkTransformer.ts

### LINKED ISSUE #427: Convert library to TypeScript
This needs to happen in steps:
- [x] Rename all the files to `*.ts` and introduce a compile step
- [x] Get the linter and formatter working
- [ ] Enable noImplicitAny and types for all src/ files
- [ ] Update the package.json to ship the types
- [ ] Enable noImplicitAny everywhere

### PR DESCRIPTION
**Pull request by joecervino**
Fixes #427 (3/21ish)

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

 🤞

### CODE REVIEW

File: src/chunktransformer.ts [hunk 1]
```diff
@@ -14,11 +14,46 @@
...
+export interface Chunk {
+  rowContents: Value;
+  commitRow: boolean;
+  resetRow: boolean;
+  rowKey?: string|Uint8Array;
+  familyName?: {value: string};
+  qualifier?: Qualifier|{value: Value};
+  timestampMicros?: number|Long;
+  labels?: string[];
+  value?: ValueOrBuffer;
```
Comment (callmehiphop) [inline src/chunktransformer.ts]: I think this will either be a string or a buffer

File: src/chunktransformer.ts [hunk 2]
```diff
@@ -45,12 +80,12 @@ export const RowStateEnum = Object.freeze({
...
-  lastRowKey;
-  state;
-  row;
-  family;
-  qualifiers;
-  qualifier;
+  lastRowKey?: Value;
+  state?: number;
+  row?: Row;
+  family?: {};
```
Comment (callmehiphop) [inline src/chunktransformer.ts]: Can we make a family type for this?

File: src/chunktransformer.ts [hunk 3]
```diff
@@ -175,10 +210,11 @@ export class ChunkTransformer extends Transform {
    * @private
    * @param {chunk} chunk chunk to validate for resetrow
    */
-  validateResetRow(chunk) {
+  validateResetRow(chunk: Chunk): void {
     const containsData = (chunk.rowKey && chunk.rowKey.length !== 0) ||
         chunk.familyName || chunk.qualifier ||
-        (chunk.value && chunk.value.length !== 0) || chunk.timestampMicros > 0;
+        (chunk.value && (chunk.value as Uint8Array).length !== 0) ||
```
Comment (callmehiphop) [inline src/chunktransformer.ts]: If we update value to be a `string|Buffer` we can get rid of this cast

File: src/chunktransformer.ts [hunk 4]
```diff
@@ -288,22 +324,24 @@ export class ChunkTransformer extends Transform {
...
    * @param {chunks} chunk chunk to process
    */
-  processNewRow(chunk) {
-    const newRowKey = Mutation.convertFromBytes(chunk.rowKey, {
+  processNewRow(chunk: Chunk): void {
+    const newRowKey = Mutation.convertFromBytes(chunk.rowKey!, {
       userOptions: this.options,
-    });
+    }) as string |
+        Buffer;
```
Comment (callmehiphop) [inline src/chunktransformer.ts]: I would just simplify this and cast to string

File: src/chunktransformer.ts [hunk 5]
```diff
@@ -354,23 +393,24 @@ export class ChunkTransformer extends Transform {
...
-        this.qualifier.value,
+        this.qualifier!.value instanceof Buffer) {
+      this.qualifier!.value = Buffer.concat([
+        this.qualifier!.value,
         chunkQualifierValue,
       ]);
     } else {
-      this.qualifier.value += chunkQualifierValue;
+      // tslint:disable-next-line no-any
+      this.qualifier!.value += chunkQualifierValue as any;
```
Comment (callmehiphop) [inline src/chunktransformer.ts]: Why any? If it isn't a buffer, it should be a string

File: src/chunktransformer.ts (hunk 1 above)
Comment (joecervino) [reply src/chunktransformer.ts]: Yeah, `valueOrBuffer` smelled funny the entire time. I should probably listen to that feeling more. Sourced it from `Mutation.convertFromBytes()` logic. Fixed 👍

File: src/chunktransformer.ts [hunk 6]
```diff
@@ -15,10 +15,44 @@
  */
 import {Transform, TransformOptions} from 'stream';
 
+import {Family} from './family';
```
Comment (callmehiphop) [inline src/chunktransformer.ts]: I don't think we want to pull this in, looks like family should be something like

```ts
interface Family {
  [qualifier: string]: Qualifier[];
}
```

Comment (joecervino) [reply src/chunktransformer.ts]: Perfect, had doubts myself, thanks Dave!

**Comment by callmehiphop** [comment]
@joecervino LGTM, let me know if you're ready for a merge

**Comment by joecervino** [comment]
@callmehiphop I was born ready to merge

**Pull request merged**